### PR TITLE
Rewrite documentation for Bedrock Scripting

### PR DESCRIPTION
**Pull request by JaylyDev**
Improve documentation for bedrock scripting begineers to understand the documents easier.

### JSON to JS
Many scripting beginners don't have the knowledge of JavaScript programming language, due to Minecraft Bedrock's add-ons primarily uses JSON. They are supposed to be expected that they to have a solid understanding of JavaScript before creating scripts for Minecraft.

Bedrock Addons #script-api channel is full of general javascript questions.  Hopefully this pr can teach beginners the fundamentals from scratch.

### Repo docs
They aren't great examples anyway, new beginners may get confused by the content they see.

### CODE REVIEW

**Comment by bot174** [comment]
Should I move these files into minecraft server folder? They don't belong to the docs root 

https://github.com/JaylyDev/ScriptAPI/blob/docs_rewrite/docs/commands_you_have_to_use.md
https://github.com/JaylyDev/ScriptAPI/blob/docs_rewrite/docs/impact_of_removing_runCommand.md

**Comment by JaylyDev** [comment]
TODO: rewrite server-net directory.

File: docs/JavaScript/README.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+- [undefined](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/undefined)
+- [Number](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number)
+- [Boolean](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Boolean)
+- [String](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String)
+- [Math](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math)
+- [Reflect](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Reflect)
+- [Symbol](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol)
+- [globalThis](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/globalThis)
+- [Date](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date)
+  > :warning: Warning: Date does not have access to the local time - the time known to the device where JavaScript is executed.
```
Comment (bot174) [inline docs/JavaScript/README.md:49]: Change Date to Script Engine

**Comment by JaylyDev** [comment]
Add async usage...?

**Comment by bot174** [comment]
Why is there a MinecraftApi directory and other minecraft api folders, should they group it in one?

**Review by bot174 — APPROVED** [review]

**Pull request merged**